packages/storage/src/database/global/proof.ts

```diff
@@ -41,11 +42,43 @@ export class ModelProof extends ModelGeneral<WithoutId<TProofRecord>> {
   public async getProof(
     database: string,
     options?: FindOptions
-  ): Promise<WithId<TProofRecord> | null> {
-    const proof = await this.collection.findOne(
+  ): Promise<TProofRecord | null> {
+    return this.collection.findOne(
       { database },
       { ...options, sort: { createdAt: -1 } }
     );
-    return proof as WithId<TProofRecord> | null;
+  }
+
+  public static async init(session?: ClientSession) {
+    const collection = ModelCollection.getInstance<TProofRecord>(
+      zkDatabaseConstant.globalProofDatabase,
+      DB.proof,
+      zkDatabaseConstant.globalCollection.proof
+    );
+    /*
+      publicInput: string[];
+      publicOutput: string[];
+      maxProofsVerified: 0 | 1 | 2;
+      proof: string;
+      createdAt?: Date;
+      database: string;
+      collection: string;
+      merkleRoot: string;
+      prevMerkleRoot: string;
+    */
+    if (!(await collection.isExist())) {
+      await collection.index({ proof: 1 }, { unique: true, session });
+      await collection.index({ database: 1 }, { session });
+      await collection.index(
+        { database: 1, collection: 1 },
+        { unique: true, session }
+      );
+      await collection.index({ merkleRoot: 1 }, { unique: true, session });
+      await collection.index({ prevMerkleRoot: 1 }, { unique: true, session });
+      await collection.index({ publicInput: 1 }, { unique: true, session });
+      await collection.index({ publicOutput: 1 }, { unique: true, session });
```

Comment:
be cautious when indexing these two array fields, make sure to index them only when needed since array indices might be expensive